extensions/VisualEditor/wikia/modules/ve/ui/dialogs/ve.ui.WikiaInfoboxInsertDialog.js

```diff
@@ -155,29 +139,117 @@ ve.ui.WikiaInfoboxInsertDialog.prototype.insertInfoboxTemplate = function () {
+ * @returns {OO.ui.OoUiSelectWidget}
+ */
+ve.ui.WikiaInfoboxInsertDialog.prototype.createInfoboxSelectWidget = function ( items ) {
+	var select = new OO.ui.SelectWidget();
+
+	select.addItems( items );
+
+	return select;
+};
+
+/**
+ * @desc creates dialog content
  * @param {Object} data Response data from API
+ * @returns {Promise}
  */
-ve.ui.WikiaInfoboxInsertDialog.prototype.showResults = function ( data ) {
-	var i,
-		items = [],
-		infoboxes = data.query ? data.query.allinfoboxes : [];
-
-	if ( infoboxes.length > 0 ) {
-		for ( i = 0; i < infoboxes.length; i++ ) {
-			items.push(
-				new OO.ui.DecoratedOptionWidget({
-					data: infoboxes[i].title,
-					label:  infoboxes[i].title
-				})
-			);
-		}
-		this.select.addItems( items );
+ve.ui.WikiaInfoboxInsertDialog.prototype.createDialogContent = function ( data ) {
+	var deferred = $.Deferred(),
+		infoboxes = data.query ? data.query.allinfoboxes : [],
+		items = infoboxes.map( this.createInfoboxItemOptionWidget );
+
+	if ( items.length > 0 ) {
+		this.select = this.createInfoboxSelectWidget( items );
+
+		this.select.connect( this, {
+			select: 'onInfoboxTemplateSelect'
+		} );
+		deferred.resolve( this.select.$element );
+	} else {
+		this.getUnconvertedInfoboxes()
+			.then(this.createEmptyState)
+			.then(deferred.resolve)
 	}
+
+	return deferred.promise();
+};
+
+/**
+ * @desc gets list of unconverted infoboxes
+ * @returns {Promise}
+ */
+ve.ui.WikiaInfoboxInsertDialog.prototype.getUnconvertedInfoboxes = function () {
+	var deferred = $.Deferred();
+
+	ve.init.target.constructor.static.apiRequest( {
+		action: 'query',
+		list: 'unconvertedinfoboxes'
+	} )
+		.done( function ( data ) {
+			deferred.resolve(
+				data.query &&
+				data.query.unconvertedinfoboxes &&
+				data.query.unconvertedinfoboxes.length > 0 ?
+					data.query.unconvertedinfoboxes :
+					[]
+			);
+		} )
+		.fail( function () {
+			deferred.resolve( [] );
+		} );
+
+	return deferred.promise();
 };
 
 /**
+ * @desc returns empty stage message
+ * @param {Array} unconvertedInfoboxes
+ * @returns {Promise}
+ */
+ve.ui.WikiaInfoboxInsertDialog.prototype.createEmptyState = function ( unconvertedInfoboxes ) {
+	var deferred = $.Deferred(),
+		noInfoboxesHTML = '<span class="insert-infobox-empty-state">' +
+			ve.msg( 'wikia-visualeditor-dialog-infobox-insert-empty-state' ) +
+			'</span>',
+		convertOldInfoboxesHTML = '<a href="' + window.location.origin +
+			'/wiki/Special:Insights/nonportableinfoboxes" class="insert-infobox-empty-state">' +
+			ve.msg( 'wikia-visualeditor-dialog-infobox-insert-empty-state-has-unconverted-infoboxes' ) +
+			'</a>';
+
+	deferred.resolve(
+		unconvertedInfoboxes.length === 0 ?
+			noInfoboxesHTML :
+		noInfoboxesHTML + convertOldInfoboxesHTML
+	);
+
+	return deferred.promise();
+};
+
+/**
+ * @desc adds html content to dialog
+ * @param {HTMLElement} $content
+ */
+ve.ui.WikiaInfoboxInsertDialog.prototype.setDialogContent = function ( $content ) {
```

Comment:
There is a mismatch between method name and method description (and what it actually do).